### ISSUE TITLE
Adding syntax highlighting for other languages (suggested fix inside)

### ISSUE THREAD

**Issue by izaid** [JuliaLang/IJulia.jl#539]
So, I know that instead of magics, IJulia suggests custom string literals -- i.e., https://github.com/JuliaLang/IJulia.jl/issues/408. What would be really great is if we could support syntax highlighting of other languages in these strings. The justification for this is that multilanguage notebooks are a somewhat popular thing. I, for example, would like to mix Julia and JavaScript every now and then. And multilanguages *do* work in IJulia, it's just that the highlighting isn't there.

However, this could be fixed as per something like this suggestion -- http://stackoverflow.com/questions/43641362/adding-syntax-highlighting-to-jupyter-notebook-cell-magic/43644349. The following modification of that actually gives JavaScript syntax highlighting to JavaScript string literals within IJulia notebooks:

```
javascript"""
require(['notebook/js/codecell'], function(codecell) {
codecell.CodeCell.options_default.highlight_modes['text/javascript'] = {'reg':[/^javascript/]} ;
  Jupyter.notebook.events.one('kernel_ready.Kernel', function(){
  Jupyter.notebook.get_cells().map(function(cell){
      if (cell.cell_type == 'code'){ cell.auto_highlight(); } }) ;
  });
});
"""
```

Would it be possible to add something like this to IJulia itself? I'd make the PR if people were happy and it was clear where to put this in the IJulia source.

**Comment by stevengj** [comment]
Wouldn't this give JavaScript syntax highlighting to the entire cell even if only the part of the cell is a JavaScript string literal?

**Comment by izaid** [comment]
Yeah, it does. I expect we could figure out a way to do highlighting of the string contents only. If that was the case, is this of interest?

**Comment by JobJob** [comment]
I think it only highlights js syntax for cells that begin with `javascript"""`, I think that's fine, I like it. Any ideas on where we could run it? 

It does work if you put just the following in ~/.jupyter/nbconfig/custom/custom.js
```
require(["notebook/js/codecell"], function(codecell) {
    codecell.CodeCell.options_default.highlight_modes["text/javascript"] = {"reg":[/^javascript/]}
})
```
But making it Julia specific would be better.

**Comment by JamesWrigley** [comment]
I'm not strongly against this, but I do feel it's possibly out of scope for IJulia so I'm going to close the issue. But if someone wants it and makes a PR I'll happily review it.
